File: Estructura_de_datos_Java/mergeSort.java
public class mergeSort {
    static void merge(int arr[], int inicio, int medio, int fin) {
        int n1 = medio - inicio + 1;
        int n2 = fin - medio;

        int[] L = new int[n1];
        int[] R = new int[n2];

        for (int i = 0; i < n1; i++) L[i] = arr[inicio + i];
        for (int j = 0; j < n2; j++) R[j] = arr[medio + 1 + j];

        int i = 0, j = 0, k = inicio;

        while (i < n1 && j < n2) {
            if (L[i] <= R[j]) arr[k++] = L[i++];
            else arr[k++] = R[j++];
        }

        while (i < n1) arr[k++] = L[i++];
        while (j < n2) arr[k++] = R[j++];
    }

    static void mergeSort(int arr[], int inicio, int fin) {
        if (inicio < fin) {
            int medio = (inicio + fin) / 2;
            mergeSort(arr, inicio, medio);
            mergeSort(arr, medio + 1, fin);
            merge(arr, inicio, medio, fin);
        }
    }

    public static void main(String[] args) {
        int arr[] = {34, 5, 62, 32, 23, 78, 1, 56};

        System.out.print("Original: ");
        for (int x : arr) System.out.print(x + " ");
        System.out.println();

        mergeSort(arr, 0, arr.length - 1);

        System.out.print("Ordenado: ");
        for (int x : arr) System.out.print(x + " ");
    }
}

File: Estructura_de_datos_C#/mergeSort.cs
using System;
namespace Estructura_de_datos_C_
{


    class Program
    {
        static void MergeSort(int[] arr, int inicio, int fin)
        {
            if (inicio < fin)
            {
                int medio = (inicio + fin) / 2;

                MergeSort(arr, inicio, medio);
                MergeSort(arr, medio + 1, fin);

                Merge(arr, inicio, medio, fin);
            }
        }

        static void Merge(int[] arr, int inicio, int medio, int fin)
        {
            int n1 = medio - inicio + 1;
            int n2 = fin - medio;

            int[] izquierda = new int[n1];
            int[] derecha = new int[n2];

            for (int i = 0; i < n1; i++)
                izquierda[i] = arr[inicio + i];
            for (int j = 0; j < n2; j++)
                derecha[j] = arr[medio + 1 + j];

            int k = inicio, i1 = 0, j1 = 0;

            while (i1 < n1 && j1 < n2)
            {
                if (izquierda[i1] <= derecha[j1])
                    arr[k++] = izquierda[i1++];
                else
                    arr[k++] = derecha[j1++];
            }

            while (i1 < n1) arr[k++] = izquierda[i1++];
            while (j1 < n2) arr[k++] = derecha[j1++];
        }

        static void Main()
        {
            int[] numeros = { 34, 7, 56, 23, 32, 5, 62 };
            Console.WriteLine("Original: " + string.Join(", ", numeros));
            MergeSort(numeros, 0, numeros.Length - 1);
            Console.WriteLine("Ordenado: " + string.Join(", ", numeros));
        }
    }
}
